lib/web_apis/relational_to_compat.es6.js

```diff
@@ -0,0 +1,84 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+require('./api_compat_data.es6.js');
+
+foam.CLASS({
+  name: 'AbstractRelationalToCompatConverter',
+  package: 'org.chromium.apis.web',
+
+  documentation: `Component that encapsulates logic for converting relational
+      Release<-->API database into a unified compat data model.`,
+
+  axioms: [foam.pattern.Singleton.create()],
+
+  requires: [
+    'org.chromium.apis.web.CompatClassGenerator',
+    'org.chromium.apis.web.CompatProperty',
+  ],
+  imports: ['info'],
+
+  properties: [
+    {
+      class: 'FObjectProperty',
+      of: 'org.chromium.apis.web.AbstractCompatClassGenerator',
+      name: 'classGenerator',
+      factory: function() { return this.CompatClassGenerator.create(); },
+    },
+  ],
+
+  methods: [
+    {
+      name: 'convert',
+      code: function(releases, apis, joins) {
+        this.info('Generating data model for ${releases.length} releases');
+        const generator = this.classGenerator;
+        const CompatData = generator.generateClass(generator.generateSpec(
+            'org.chromium.apis.web.generated', 'CompatData', releases));
+        this.info('Data model generated');
+
+        const compatProps = CompatData.getAxiomsByClass(this.CompatProperty);
+        const setCompatBit = (data, releaseId) => {
```

Comment:
Maybe move this closer to usage, or just inline it because it's only used in one place.

Reply:
Done